framework/src/base/SubProblem.C

```diff
@@ -46,6 +46,60 @@ SubProblem::SubProblem(const InputParameters & parameters)
 
 SubProblem::~SubProblem() {}
 
+TagID
+SubProblem::addVectorTag(TagName tag_name)
+{
+  if (_vector_tag_name_to_tag_id.find(tag_name) == _vector_tag_name_to_tag_id.end())
```

Comment:
It is better to have something like
```
auto it = _vector_tag_name_to_tag_id.find(tag_name);
if (it == _vector_tag_name_to_tag_id.end())
{
   ...
   return tag_id;
}
else
  return it->second;
```
This piece of code is not efficiency critical but set a bad example of using map.